msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -883,6 +1030,37 @@ public List getUserAuthoredStatistics2(){
 		return statistics;
 	}
 	
+	/**
+	 * Filters out messages associated with anonymous topics for which the user is not permitted to see IDs
+	 */
+	private List<Message> filterToAnonAwareMessages(List<Message> messages)
+	{
+		if (!isAnonymousEnabled() || m_displayAnonIds)
+		{
+			return messages;
+		}
+
+		List<Message> filtered = new ArrayList<Message>();
+		// Map topics to value of 'isUseAnonymousId(Topic)' to prevent redundant queries
+		Map<Topic, Boolean> topicToUseAnon = new HashMap<Topic, Boolean>();
+		for (Message message : messages)
+		{
+			Topic topic = message.getTopic();
```

Comment:
I've seen that you're null checking the `Topic` object in other places, perhaps here as well? Also, what about the `messages` list being passed into this method?